dstep/translator/Context.d

```diff
@@ -198,9 +219,72 @@ bool shouldBeAnonymous(Context context, Cursor cursor)
 }
 
 /**
-  * Returns true, if cursor is in the global scope.
-  */
+ * Returns true, if cursor is in the global scope.
+ */
 bool isGlobal(Cursor cursor)
 {
     return cursor.semanticParent.kind == CXCursorKind.CXCursor_TranslationUnit;
 }
+
+/**
+ * The collectTypeNames function scans the whole AST of the translation unit and produces
+ * a set of the type names in global scope.
+ *
+ * The type names are required for the parsing of C code (e.g. macro definition bodies),
+ * as C grammar isn't context free.
+ */
+
+bool[string] collectTypeNames(TranslationUnit translUnit)
```

Comment:
I don't like the second suggestion with `TypeImpl`. I'm leaning towards the first suggestion with classes. The last suggestion isn't so bad either. The last one has the advantage that we don't need to come up with names for two new classes (I'm not a big fan of `UserType`).